tests/all/component_model/call_hook.rs

```diff
@@ -70,6 +71,87 @@ fn call_wrapped_func() -> Result<()> {
     Ok(())
 }
 
+// Crate a synchronous Func, call it directly:
+#[test]
+fn call_wrapped_func_with_realloc() -> Result<()> {
+    let wat = format!(
+        r#"
+        (component
+            (import "f" (func $f (param "list" (list u8))))
```

Comment:
Could you shift this to being a result of a list? As-is this doesn't require a realloc, but as a return value it'll require a realloc